packages/at_persistence_secondary_server/lib/src/model/at_metadata_builder.dart

```diff
@@ -26,11 +26,25 @@ class AtMetadataBuilder {
       String? encoding}) {
     newAtMetaData ??= AtMetaData();
     atMetaData = newAtMetaData;
-    atMetaData!.createdAt ??= currentUtcTime;
+    // createdAt indicates the date and time of the key created.
+    // For a new key, the currentDateTime is set and remains unchanged
+    // on an update event.
+    (existingMetaData?.createdAt == null)
+        ? atMetaData!.createdAt = currentUtcTime
+        : atMetaData!.createdAt = existingMetaData?.createdAt;
     atMetaData!.createdBy ??= atSign;
     atMetaData!.updatedBy = atSign;
+    // updatedAt indicates the date and time of the key updated.
+    // For a new key, the updatedAt is same as createdAt and on key
+    // update, set the updatedAt to the currentDateTime.
     atMetaData!.updatedAt = currentUtcTime;
     atMetaData!.status = 'active';
+    // The version indicates the number of updates a key has received.
+    // Version is set to 0 for a new key and for each update the key receives,
+    // the version increases by 1
+    (existingMetaData?.version == null)
```

Comment:
Can you check in the callers whether existingMetaData is populated from hive. In case if some callers fails to pass existingMetaData for an existing key, then version will be set to 0.

Reply:
Thank you Murali, Except in "putAll" method, existingMetadata is fetched and passed to AtMetadataBuilder. I have added for "putAll" method as well. 
Units are also updated to put, putMeta and putAll method to update the key and verify version is set to 1